cmd/common/display/message.go

```diff
@@ -107,16 +107,18 @@ func (r *RespTxQuery) MarshalJSON() ([]byte, error) {
 		TxResult: r.Msg.TxResult,
 	}
 	// Always try to serialize to verify hash, but only show raw if requested.
-	raw, err := r.Msg.Tx.MarshalBinary()
-	if err != nil {
-		out.Warn = "ERROR encoding transaction: " + err.Error()
-	} else {
-		if r.WithRaw {
-			out.Raw = hex.EncodeToString(raw)
-			hash := sha256.Sum256(raw)
-			if !bytes.Equal(hash[:], r.Msg.Hash) {
-				out.Warn = fmt.Sprintf("HASH MISMATCH: requested %x; received %x",
-					r.Msg.Hash, hash)
+	if r.Msg.Tx != nil {
+		raw, err := r.Msg.Tx.MarshalBinary()
+		if err != nil {
+			out.Warn = "ERROR encoding transaction: " + err.Error()
+		} else {
+			if r.WithRaw {
```

Comment:
So I guess I'm the culprit here, but let's pls revise to `} else if r.WithRaw {`
The indent is getting kind deep now.